librosa/util/utils.py

```diff
@@ -627,8 +632,19 @@ def normalize(S, norm=np.inf, axis=0):
     else:
         raise ParameterError('Unsupported norm: {}'.format(repr(norm)))
 
+    # get the vectors-ids smaller than the threshold
+    small_values_idx = length < threshold
+
     # Avoid div-by-zero
-    length[length < SMALL_FLOAT] = 1.0
+    length[small_values_idx] = 1.0
+
+    if axis == 1:
```

Comment:
Is there a way to index an array based on the `axis` parameter?

Otherwise we need an `if` for each axis to get the indexing right.